backend/src/server/routes/v1/secret-sharing-router.ts

```diff
@@ -74,12 +74,66 @@ export const registerSecretSharingRouter = async (server: FastifyZodProvider) =>
       }
     },
     handler: async (req) => {
-      const sharedSecret = await req.server.services.secretSharing.getActiveSharedSecretById({
+      const sharedSecret = await req.server.services.secretSharing.getPasswordlessSecretByID({
         sharedSecretId: req.params.id,
         hashedHex: req.query.hashedHex,
         orgId: req.permission?.orgId
       });
+
+      if (!sharedSecret) return undefined;
+
+      return {
+        encryptedValue: sharedSecret.encryptedValue,
```

Comment:
The encrypted value must not return if password is provided because if it does defeats the whole purpose of password protection

Reply:
the encrypted value is never returned unless it's not password protected and we only return it so it can be deciphered on the client.

Reply:
I am a bit confused. From the code i see the encrypted value is returned irrespective of password protected. Can you please point me where this is getting stripped off?

Reply:
@akhilmhdh 

The code highlighted above returns a passwordless encrypted secret to be deciphered on the client. The password validation happens later. 

At this point no password has been sent to the server. That's on initial load.